pkg/mimirpb/mimir.pb.go

```diff
@@ -7584,7 +7584,6 @@ func (m *WriteRequest) Unmarshal(dAtA []byte) error {
 			m.Metadata = append(m.Metadata, metadata)
 		}
 		m.rw2symbols.releasePages()
-		m.unmarshalFromRW2 = false
```

Comment:
What does this change signify?

Reply:
Before this PR I cleaned up almost everything that was used to decode the data buffer as remote-write 2, including this marker. However at the place where I need the information about the protocol `(d *Distributor) metricsMiddleware(next PushFunc) PushFunc`  , I only get `WriteRequest`, not `PreallocWriteRequest` so I don't have the info anymore. Easiest was to not clear this marker - doesn't hurt, cause it's not allocation.